research/zkemail/README.md

```diff
@@ -0,0 +1,273 @@
+
+The ZK Email team mitigated the issue by changing how the email address is extracted. In particular, the circuit [reverses](https://github.com/zkemail/zk-regex/blob/7002a2179e076449b84e3e7e8ba94e88d0a2dc2f/packages/circom/circuits/common/email_addr_with_name_regex.circom#L13-L15) the From header, looks for the [first occurrence of angle brackets](https://github.com/zkemail/zk-regex/blob/7002a2179e076449b84e3e7e8ba94e88d0a2dc2f/packages/circom/circuits/common/reversed_bracket_regex.circom#L5), and extracts what is inside of them. Note, however, that the number of possible parser discrepancies is practically unlimited, especially given the variety of parsers and the fact that their code changes over time. Thus, it is crucial to have defense-in-depth strategies like timelocks for ZK Email actions.
+
+
+## Enigmatic 255-decimal
+
+The cornerstone part of the ZK Email system is the ZK regex compiler. It empowers a ZK circuit to support the processing of input signals with beloved and cherished regular expressions. The ZK regexes allow not only to constrain the input to satisfy the regex but additionally to produce a revealing array, exposing the part of the input matching specific sub-regexes.
+
+Let’s look at the concrete example of a regex fed to the ZK regex compiler. This one comprises several sub-regexes and should match the whole `dkim-signature` header in an email but reveal only the timestamp value for the `;t=ts_value` tag in the header.
+
+```
+{
+  "parts": [
+    {
+      "is_public": false,
+      "regex_def": "(\r\n|^)dkim-signature:"
+    },
+    {
+      "is_public": false,
+      "regex_def": "([a-z]+=[^;]+; )+t="
+    },
+    {
+      "is_public": true,
+      "regex_def": "[0-9]+"
+    },
+    {
+      "is_public": false,
+      "regex_def": ";"
+    }
+  ]
+}
+```
+
+The ZK Email relies on regular expressions prefixed with `(\r\n|^)header_name:` for extracting various headers from an email, such as the `from`, `subject`, `dkim-signature` headers.
+
+When we compile a regular expression with the compiler, it generates a Circom circuit that embodies a Deterministic Finite Automaton (DFA) satisfying the regular expression. Any regular expression can be converted into an equivalent DFA. An input string matches the regular expression if it transitions the DFA to the accept state.
+
+Let’s have the `^a` regular expression as a toy example. When generating a circuit for the regular expression, the compiler makes a DFA with three states, with state `2` being the accepted state.
```

Comment:
Add a link to the Regex visualizer with DFA picture: https://zkregex.com/min_dfa?regex=XmE=

Reply:
fixed